File: classfields/counters.test.js
const counters = require('./counters');

it('должен вернуть данные за все время и сегодня, если есть данные', () => {
    expect(counters({
        counters: { all: 20, daily: 10 },
    })).toEqual('20 (10 сегодня)');
});

it('должен вернуть данные за все время, если нет данных за сегодня', () => {
    expect(counters({
        counters: { all: 20, daily: 0 },
    })).toEqual('20');
});

it('должен вернуть пустую строку, если нет данных', () => {
    expect(counters({
        counters: { all: 0 },
    })).toEqual('');
});

File: classfields/test/scala/ru/auto/api/CountersTest.scala
package ru.auto.api

import org.scalatest.funsuite.AnyFunSuite
import ru.auto.api.util.Protobuf
import org.scalatest.matchers.should.Matchers._
import ru.auto.api.CounterModel.AggregatedCounter
import ru.auto.api.util.JsonMatchers._

/**
  * Author: Vladislav Dolbilov (darl@yandex-team.ru)
  * Created: 18.02.17
  */
class CountersTest extends AnyFunSuite {

  test("always serialize zero values") {
    val counters = AggregatedCounter
      .newBuilder()
      .setAll(0)
      .setDaily(0)
      .setPhoneAll(0)
      .setPhoneDaily(0)
      .build()

    Protobuf.toJson(counters) should matchJson("""{
        |  "all": 0,
        |  "daily": 0,
        |  "phone_all": 0,
        |  "phone_daily": 0
        |}""".stripMargin)
  }
}
